### PR TITLE
Follow up Cere tweaks

### PR DESCRIPTION
**Pull request by S34NW**
## What Does This PR Do
Minor fixes that are required following the previous massive PR.

Adds plating under doors and shutters in cargo.
Adds more lights to library and bar.
Fixes the floor pattern in service south.
Fixes https://github.com/ParadiseSS13/Paradise/issues/20716
Fixes a longstanding bug with autoname cameras

## Why It's Good For The Game
Minor issues of no real consequence, but still annoying.

## Testing
Eyeballs

## Changelog
:cl:
fix: Fixed minor issues that persist on cere station
/:cl:

### CODE REVIEW

**Comment by S34NW** [comment]
> * Detective's Request Console is missing `department` and `departmentType`, making it show up as "Unknown's Request Console"-
>       imagen
> 
>     * Guh. Who dropped the sign on the hallway...
>       imagen
>       -Cameras in Fitness Room, Bar, Kitchen and Primary Cryogenic Dorms break the camera console with an empty name and image. I would say it's because they are the `/autoname` variant, which seems superfluous as the basic security cams already name themselves... But there are others (see pic) of this variant that _do_ work? Strange behaviour.
>       imagen
> 
>     * Chute windoor in Arrivals Security checkpoint lacks detective access.

Fixed all of these, the camera one was due to a longstanding bug in these cameras that has never been fixed (area names included `\improper` which broke things). It now works as intended.

File: code/game/machinery/camera/camera_presets.dm [hunk 1]
```diff
@@ -38,21 +38,24 @@
...
+
+	for(var/obj/machinery/camera/autoname/C in A.contents)
+		if(C == src)
+			continue
+		if(C.number)
+			number = max(number, C.number + 1)
+
+	c_tag = "[sanitize(A.name)] #[number]"
 
+	..() // We do this here so the camera is not added to the cameranet until it has a name.
```
Comment (farie82) [inline code/game/machinery/camera/camera_presets.dm:58]:
```suggestion
	return ..() // We do this here so the camera is not added to the cameranet until it has a name.
```

File: code/game/machinery/camera/camera_presets.dm [hunk 2]
```diff
@@ -38,21 +38,24 @@
...
-		for(var/obj/machinery/camera/autoname/C in GLOB.machines)
-			if(C == src)
-				continue
-			var/area/CA = get_area(C)
-			if(CA.type == A.type)
-				if(C.number)
-					number = max(number, C.number + 1)
-		c_tag = "[A.name] #[number]"
+	if(!A)
+		number = rand(1, 100)
```
Comment (farie82) [inline code/game/machinery/camera/camera_presets.dm:46]: Might also want to put down a stacktrace here. I don't think this should be expected behaviour ever

File: code/game/machinery/camera/camera_presets.dm [hunk 3]
```diff
@@ -38,21 +38,24 @@
...
+		c_tag = "Unknown #[number]"
+		return ..()
+
+	for(var/obj/machinery/camera/autoname/C in A.contents)
```
Comment (farie82) [inline code/game/machinery/camera/camera_presets.dm:50]: IIRC looping over an area's contents was super expensive/inefficient. Might be worth benchmarking to see if it's cheaper to loop over the `GLOB.cameranet.cameras` list instead and do an area check

Comment (S34NW) [reply code/game/machinery/camera/camera_presets.dm:50]: Is this true for typecasted loops? I totally forget

Comment (farie82) [reply code/game/machinery/camera/camera_presets.dm:50]: @AffectedArc07 you might know a bit more about this

Comment (AffectedArc07) [reply code/game/machinery/camera/camera_presets.dm:51]: Its on `Initialize()`. Not particularly an issue.

Coderbus would be best bet for knowledge.

Comment (farie82) [reply code/game/machinery/camera/camera_presets.dm:51]: Should be fine to go then

**Review by farie82 — APPROVED** [review]

**Pull request merged**